src/pykeen/utils.py

```diff
@@ -1777,6 +1780,61 @@ def isin_many_dim(elements: torch.Tensor, test_elements: torch.Tensor, dim: int
     return counts[inverse[: elements.shape[dim]]] > 1
 
 
+def minmax_normalize(array: np.ndarray) -> np.ndarray:
+    """
+    Normalize the given array using min-max normalization.
+
+    :param array:
+        array to normalize
+
+    :return:
+        normalized array
+    """
+    max_lit, min_lit = np.max(array, axis=0), np.min(array, axis=0)
+    return (array - min_lit) / (max_lit - min_lit + 1e-8)
+
+
+def filter_triples_by_relations(triples: LabeledTriples, min_occurrences: int = 0, regex: str = "") -> LabeledTriples:
+    """
+    Filter triples based on their relations' occurrences and labels.
+
+    :param triples:
+        triples to filter
+    :param min_occurrences:
+        minimum number of occurrences for a relation to be kept
+    :param regex:
+        regex to match with relation's label for the relation to be kept
+
+    :return:
+        filtered triples
+    """
+    relations = triples[:, 1]
+    relation_counts = np.unique(relations, return_counts=True)
+    relation_counts = dict(zip(relation_counts[0], relation_counts[1]))
+
+    filtered_triples = []
+    for h, r, t in triples:
+        if relation_counts[r] < min_occurrences or not re.search(regex, str(r)):
+            continue
+        filtered_triples.append((h, r, t))
+
+    filtered_triples = np.array(filtered_triples, dtype=np.dtype("O"))
+
+    return filtered_triples
```

Comment:
I would prefer to

1. have a method with works on mapped triples
2. have some wrapper around it which determines the relation ids, whose label matches the regex.

I'll add a PR to your PR with a suggested implementation.

Reply:
-> https://github.com/AntonisKl/pykeen/pull/1

It loses a bit of generality, but lets the whole preprocessing code run vectorized.